env0/resource_project_policy_test.go

```diff
@@ -0,0 +1,94 @@
+package env0
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/golang/mock/gomock"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestUnitPolicyResource(t *testing.T) {
+	resourceType := "env0_project_policy"
+	resourceName := "test"
+	accessor := resourceAccessor(resourceType, resourceName)
+
+	policy := client.Policy{
+		Id:                   "id0",
+		ProjectId:            "project0",
+		NumberOfEnvironments: 1,
+	}
+
+	updatedPolicy := client.Policy{
+		Id:                   policy.Id,
+		ProjectId:            policy.ProjectId,
+		NumberOfEnvironments: 0,
+	}
+
+	testCase := resource.TestCase{
+		Steps: []resource.TestStep{
+			{
+				Config: resourceConfigCreate(resourceType, resourceName, map[string]interface{}{
+					"project_id":             policy.ProjectId,
+					"number_of_environments": policy.NumberOfEnvironments,
+				}),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(accessor, "project_id", policy.ProjectId),
+					resource.TestCheckResourceAttr(accessor, "number_of_environments", fmt.Sprintf("%d", policy.NumberOfEnvironments)),
```

Comment:
Can we please also validate all other fields? (relevant both for create & update)